### PR TITLE
Fix pressure bc energy/gradient discrepancy and add test

### PR DESCRIPTION
**Pull request by arvigj**

### CODE REVIEW

File: src/polyfem/assembler/RhsAssembler.cpp [hunk 1]
```diff
@@ -729,7 +729,7 @@ namespace polyfem
 			}
 		}
 
-		double RhsAssembler::compute_energy(const Eigen::MatrixXd &displacement, const std::vector<LocalBoundary> &local_neumann_boundary, const Density &density, const int resolution, const double t) const
+		double RhsAssembler::compute_energy(const Eigen::MatrixXd &displacement, const Eigen::MatrixXd &displacement_prev, const std::vector<LocalBoundary> &local_neumann_boundary, const Density &density, const int resolution, const double t) const
```
Comment (teseoch) [inline src/polyfem/assembler/RhsAssembler.cpp:732]: do we need displacement?

so we use the prev for the normals, but displacement for the rest?

Comment (arvigj) [reply src/polyfem/assembler/RhsAssembler.cpp:732]: Yes, so the normal computation for the pressure is lagged but the energy needs the current displacement as the gradient is constant.

**Review by teseoch — APPROVED** [review]

**Comment by teseoch** [comment]
Thanks!!! lets hope it is the last problem related to pressure and Neumann :P

**Comment by arvigj** [comment]
Well so I'm not happy with pressure from a physical modeling point of view, but I have a fix for that on the horizon. This should at least make the math self consistent.

**Pull request merged**